### ISSUE TITLE
Too filter partial swears.

### ISSUE THREAD

**Issue by tonyram57** [richardfrost/AdvancedProfanityFilter#130]
Example I see a lot of swears like this.  Partial masking that should be censored.

![2019-04-04_105838](https://user-images.githubusercontent.com/39742654/55565885-adc00000-56c8-11e9-8277-d7ff8ca2eaed.png)

Also request some racial slurs like "retarded" and "fag" (no offensive to anyone) be censored by default.
I know this words can be added but still should be filtered by default.

**Comment by tonyram57** [comment]
Okay I figured out how to remove words and add those words.  I think filtering "Jesus" and "Christ" is crazy as filtering "hell" but I know how to remove those words now.

Question though when the extension is updated I hope it does not add those filtered words again.

**Comment by richardfrost** [comment]
Hello @tonyram57, 

I understand that the default word list isn't for everyone. I have some ideas to hopefully make it a little better in the future, but for now, tailoring the config to your liking is the best solution to having the filter work for you. The defaults shouldn't override your config (unless you happen to remove all the words from the word list, then it will revert to the defaults).

To safeguard your preferences, I always recommend backing up your config (Options -> Config -> Export), just in case something goes wrong.

In regards to the partial masking, unfortunately I don't think I can anticipate every variation of spelling for words to filter. If you see one, you can add it to the filter by highlighting the desired word/phrase and then opening the context menu (usually right-click) and then select Advanced Profanity Filter -> Add selection to filter.

If you have more suggestions for a default words, please let me know and I'll consider adding it. I keep a list of word suggestions and try to add the most requested words to the defaults. I'll add your 2 suggestions to the list and definitely consider adding them to the next update. As was mentioned above, Even if they do get added to defaults, they won't be applied for existing users. If they do get added as a  default, I'll document it in the release notes so users can add them to their own custom config if they want.

**Comment by richardfrost** [comment]
From your feedback and from others, I'm removing a few of the words from the defaults, as it is probably too strict fro most people. I have also added a couple new defaults based on your input. I do hope sometime to revamp the settings, but that won't be for a while. The changes take effect in `2.1.2`, which will be out shortly.

**Comment by notanewbie** [comment]
I'd like to recommend changes to a few of the substitutions BTW, some of them just feel odd to me:
bad--- --> bad boy (was "cool")
bast--d --> guy (was "jerk")
b---h --> whiner (was "jerk")
c---sucker --> bootlick or suckup (was "idiot")
c--t --> idiot, hard, or politician (was "explative") [hard because of the various contexts]
f-- --> gay (was "slur")
f---ot --> gay (was "slur")
f--s --> gay (was "slur")
f--k --> dang, screw, or forget (was "fudge")
g--d--nit --> ah dangit (was goshdangit)
jack--- --> jerk (was "idiot")
n---a --> bruh (was "ethnic slur")
n---er --> man (was "ethnic slur")
n---ers --> people (was "ethnic slur")
t--s --> boobs or chest (was "explative")
tw-- --> dumbo (was "explative")

Also how come expletive is always mispelled "explative' in your extension?

**Comment by richardfrost** [comment]
Thanks so much for the input! I definitely appreciate your thoughts on these. Several of these you can tell I just kind of put in some place holders and hoped to get feedback on what people wanted. Again, these are just the defaults, so anyone is able to adjust theirs to their liking.

The misspelling has been addressed and will be fixed in the next version #139. I ended up getting an export of a user's config and it was misspelled once in there, and from there I think it just got copied/pasted everywhere else as a placeholder. Sorry for the trouble.

**Comment by richardfrost** [comment]
I really like several of your suggestions @notanewbie, and I'm ready to set them to be the defaults:

```
c---sucker --> suckup (was "idiot")
f-- --> gay (was "slur")
f---ot --> gay (was "slur")
f--s --> gays (was "slur")
jack--- --> jerk (was "idiot")
n---a --> bruh (was "ethnic slur")
n---er --> man (was "ethnic slur")
n---ers --> people (was "ethnic slur")
t--s --> chest (was "explative")
tw-- --> dumbo (was "explative")
```

I think I like simplifying this one:
```
g--d--nit --> ah dangit (was goshdangit) = dangit
```

The rest of the suggestions I'm not ready to adjust yet. It is much harder for words that have multiple meanings and no clear substitution. I have a set of sentences/phrases from the web I run the filter through to see if substitutions make sense, and unfortunately these ones fail about the same, or more than the current ones. I did add a couple notes surrounded by square brackets below. I'm open to more suggestions.

Notes: I don't like the current substitutions, but I feel like these new ones don't really make them better.
```
b---h --> whiner (was "jerk") [Doesn't fit all uses.]
bad--- --> bad boy (was "cool") [Gender can make this confusing: "she's a bad boy."]
bast--d --> guy (was "jerk") [Gender can make this confusing: "She's a guy child."]
c--t --> idiot, hard, or politician (was "explative") (hard because of the various contexts) [Maybe leave this one as "expletive"? That would be the only one.]
f--k --> dang, screw, or forget (was "fudge") [Doesn't fit all uses. What about "F"? Or maybe "F---", "freak", "flip", "frick" are some of the others I've heard]
```

I have a "The Good Place" word list that I use sometimes, and I wish it was that simple on some of these, like bench and fork. Maybe I should just use those?

<details>
  <summary>The Good Place Words (Warning: Profanity)</summary>

```json
{
  "words": {
    "ass": { "matchMethod": 0, "repeat": true, "sub": "ash" },
    "asses": { "matchMethod": 0, "repeat": false, "sub": "ashes" },
    "asshole": { "matchMethod": 1, "repeat": true, "sub": "ashhole" },
    "bastard": { "matchMethod": 1, "repeat": true, "sub": "badword" },
    "bitch": { "matchMethod": 1, "repeat": true, "sub": "bench" },
    "cunt": { "matchMethod": 1, "repeat": true, "sub": "expletive" },
    "dammit": { "matchMethod": 1, "repeat": true, "sub": "dangit" },
    "damn": { "matchMethod": 1, "repeat": true, "sub": "dang" },
    "dick": { "matchMethod": 0, "repeat": true, "sub": "dink" },
    "dumbass": { "matchMethod": 0, "repeat": true, "sub": "dumbash" },
    "fuck": { "matchMethod": 1, "repeat": true, "sub": "fork" },
    "god": { "matchMethod": 0, "repeat": false, "sub": "gosh" },
    "hell": { "matchMethod": 0, "repeat": true, "sub": "heck" },
    "piss": { "matchMethod": 1, "repeat": true, "sub": "pee" },
    "pissed": { "matchMethod": 0, "repeat": true, "sub": "ticked" },
    "shit": { "matchMethod": 1, "repeat": true, "sub": "shirt" },
    "slut": { "matchMethod": 1, "repeat": true, "sub": "tramp" },
    "tits": { "matchMethod": 1, "repeat": true, "sub": "chest" },
    "whore": { "matchMethod": 1, "repeat": true, "sub": "tramp" }
  }
}

```
</details>

One thought I'm having right now is changing the default filter mode from substitute back to censor (what it was orginally). It is easier to make a good replacement with it. I did have a few people request to make substitute the default, but maybe that was a hasty decision.
